tensorflow_addons/rnn/cell.py

```diff
@@ -619,3 +619,176 @@ def get_config(self):
 
         ln_config["layernorm_epsilon"] = ln_config.pop("epsilon")
         return dict(list(cell_config.items()) + list(ln_config.items()))
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class ESNCell(keras.layers.AbstractRNNCell):
+    """Echo State recurrent Network (ESN) cell.
+
+    This implements the recurrent cell from the paper:
+        H. Jaeger
+        "The "echo state" approach to analysing and training recurrent neural networks". GMD Report148, German National Research Center for Information Technology, 2001.
+
+    Arguments:
+        units: Positive integer, dimensionality in the reservoir.
+        connectivity: Float between 0 and 1.
+            Connection probability between two reservoir units.
+            Default: 0.1.
+        leaky: Float between 0 and 1.
+            Leaking rate of the reservoir.
+            If you pass 1, it's the special case the model does not have leaky integration.
```

Comment:
Actually, if "leaky = 0", it will always return the recurrent state. As the weights of the ESN are not trainable, it will return always the same value, the initial state. Thus, it will not have much sense. In the opposite scenario, when "leaky = 1", call returns the output itself as it would do it if there was no leaky integration.